hibernate-core/src/main/java/org/hibernate/type/descriptor/java/AbstractTemporalJavaType.java

```diff
@@ -60,6 +64,13 @@ private <X> TemporalJavaType<X> forMissingPrecision(TypeConfiguration typeConfig
 		return (TemporalJavaType) this;
 	}
 
+	public static Time millisToSqlTime(long millis) {
+		final LocalTime localTime = Instant.ofEpochMilli( millis ).atZone( ZoneId.systemDefault() ).toLocalTime();
+		final Time time = Time.valueOf( localTime );
+		time.setTime( time.getTime() + localTime.getNano() / 1_000_000 );
```

Comment:
Is it always correct to assume the provided epoch value is in the `ZoneId.systemDefault()` time zone? I'm afraid this might cause problems. Also having to restore milliseconds using `getNano()` is unfortunate.

I would much rather prefer an implementation similar to `JdbcDateJavaType#toDateEpoch`, e.g. `toTimeEpoch` that instead sets the `YEAR`, `MONTH` and `DATE` to `1970-01-01` and derives the correct time-only milliseconds value from that.